### PR TITLE
Refactor the defination of transport topics

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: myan <myan@redhat.com>

### CODE REVIEW

File: agent/pkg/status/controller/generic/generic_interface.go [hunk 1]
```diff
@@ -26,12 +29,12 @@ type EventEmitter interface {
...
 	ToCloudEvent() *cloudevents.Event
 
-	// to assert whether emit the current cloudevent
-	Emit() bool
-
 	// topic
 	Topic() string
 
+	// to assert whether send the current cloudevent
+	PreSend() bool
```
Comment (clyang82) [inline agent/pkg/status/controller/generic/generic_interface.go:36]: maybe `ShouldSend()`. `presend` and `postsend` is a pair. And should have the same return value.

File: agent/pkg/status/controller/event/local_root_policy_emitter.go [hunk 2]
```diff
@@ -52,6 +52,38 @@ func NewLocalRootPolicyEmitter(ctx context.Context, c client.Client) *localRootP
...
+		return false
+	}
+
+	policy, ok := policyEventPredicate(h.ctx, obj, h.runtimeClient, h.log)
+
+	return ok && !utils.HasAnnotation(policy, constants.OriginOwnerReferenceAnnotation) &&
+		!utils.HasItemKey(policy.GetLabels(), constants.PolicyEventRootPolicyNameLabelKey)
+}
+
+func policyEventPredicate(ctx context.Context, obj client.Object, c client.Client, log logr.Logger) (
```
Comment (clyang82) [inline agent/pkg/status/controller/event/local_root_policy_emitter.go:67]: Do we watch other events or only watch policy events now?

File: operator/pkg/controllers/addon/manifests/templates/hostedagent/multicluster-global-hub-hosting-agent-deployment.yaml [hunk 3]
```diff
@@ -36,7 +36,6 @@ spec:
             - --lease-duration={{.LeaseDuration}}
             - --renew-deadline={{.RenewDeadline}}
             - --retry-period={{.RetryPeriod}}
-            - --kubernetes-event-exporter-config=/kube-event/config.yaml
```
[the remark this answers is not in the document]
Comment (yanmxa) [reply operator/pkg/controllers/addon/manifests/templates/hostedagent/multicluster-global-hub-hosting-agent-deployment.yaml:39]: Remove the kube event-exporter from this: https://github.com/stolostron/multicluster-global-hub/pull/815

File: agent/pkg/status/controller/generic/generic_interface.go (hunk 1 above)
Comment (yanmxa) [reply agent/pkg/status/controller/generic/generic_interface.go]: Thanks! I will consider these changes and naming in another pull request. For now, I will remove them from here.

File: agent/pkg/status/controller/event/local_root_policy_emitter.go (hunk 2 above)
Comment (yanmxa) [reply agent/pkg/status/controller/event/local_root_policy_emitter.go]: We watch all the events and only filter the policy events to process

**Review by clyang82 — APPROVED** [review]

**Pull request merged**